### PR TITLE
update: Continuous integration to update image with Flux

### PR DESCRIPTION
**Pull request by ybezsonov**
#### Which issue(s) this PR fixes:
#511
#389  

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test` or `make e2e-test` and it was successful

  EKS Workshop
    Automation
      Unknown
        Flux
          ✔ Flux (195788ms)
          ✔ Accessing AWS CodeCommit (83ms)
          ✔ Cluster bootstrap (22763ms)
          ✔ Deploying an application (84230ms)
          ✔ Continuous Integration and GitOps (50912ms)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by csantanapr** [comment]
This is ready for review @allamand @tsahiduek could you review?

**Comment by csantanapr** [comment]
@niallthomson could you run an end2end test pipeline on this PR?

File: website/docs/automation/gitops/flux/ci.md
Comment (allamand) [inline website/docs/automation/gitops/flux/ci.md:27]: Maybe add a note on why we do this, put a link on Graviton on it's advantage ?

Comment (allamand) [reply website/docs/automation/gitops/flux/ci.md:27]: Also make a ref to the terraform code so user can better understand how a single buildspec.yml result in 2 amd and arm build in the pipeline

**Review by allamand — CHANGES_REQUESTED** [review]

File: manifests/modules/automation/gitops/flux/buildspec-manifest.yml
Comment (ybezsonov) [reply manifests/modules/automation/gitops/flux/buildspec-manifest.yml:23]: One is for tag latest, which I use for the initial image in manifest, when the app is deployed.
Another is for tag iTtimestapm, which is used for incremental builds and FluxImageUpdater.
ImageManifest combine together reference to latest-arm64 and latest-amd64 images.

File: website/docs/automation/gitops/flux/ci.md
Comment (ybezsonov) [reply website/docs/automation/gitops/flux/ci.md:27]: We have a link to terraform in the very beginning of the lab and also the diagram. I agree that we need more info on Graviton, but may be in the separate lab when we deploy built images to arm64 nodes.

@niallthomson @csantanapr Should we simplify this lab and build only x64 images and than introduce multi-arch separately with deploy to arm64 nodes?

File: website/docs/automation/gitops/flux/ci.md
Comment (ybezsonov) [reply website/docs/automation/gitops/flux/ci.md]: fixed

File: manifests/modules/automation/gitops/flux/buildspec-manifest.yml
Comment (allamand) [reply manifests/modules/automation/gitops/flux/buildspec-manifest.yml:23]: latest is not a best practices to uses, do we uses the latest somewhere after ?

Comment (ybezsonov) [reply manifests/modules/automation/gitops/flux/buildspec-manifest.yml:23]: We don't use latest in deployment, it is rather a label to show that this image with tag iTimestapm is the latest. We can comment out or remove lines which use/build latest. But sometimes it is useful to have it, because it can be used as a trigger in the CodePipeline to start some other actions when a new image has been built.

File: website/docs/automation/gitops/flux/ci.md
Comment (csantanapr) [reply website/docs/automation/gitops/flux/ci.md:27]: Lets keep arm64 image and add short sentence with a link to a graviton as informational

File: website/docs/automation/gitops/flux/ci.md
Comment (csantanapr) [inline website/docs/automation/gitops/flux/ci.md:173]:
```suggestion
$ wait-for-lb $UI_URL
```

There is already a `wait-for-lb` to make it simpler and cleaner to wait for url to be up

File: website/docs/automation/gitops/flux/ci.md
Comment (csantanapr) [inline website/docs/automation/gitops/flux/ci.md:62]:
```suggestion
While we are waiting for pipeline to create new images (approximately 4 minutes) , let's [automate image updates to Git](https://fluxcd.io/flux/guides/image-update/) using Flux Image Automation Controller.
```

File: website/docs/automation/gitops/flux/ci.md
Comment (csantanapr) [inline website/docs/automation/gitops/flux/ci.md:226]:
```suggestion
$ wait-for-lb $UI_URL
```

**Review by csantanapr — APPROVED** [review]
Minor nits, otherwise good to merge

**Comment by csantanapr** [comment]
Thank you so much @ybezsonov 🎉 
I was able to test it and it works for me 
@niallthomson please merge this when you have a chance

File: manifests/modules/automation/gitops/flux/.workshop/terraform/addon.tf
Comment (niallthomson) [inline manifests/modules/automation/gitops/flux/.workshop/terraform/addon.tf:135]: Can you add a random suffix to this? Currently if this lab fails to clean up for some reason it will block any subsequent test runs without manual cleanup.

**Review by niallthomson — CHANGES_REQUESTED** [review]
A great PR as usual @ybezsonov I have made 1 request for a change I feel is needed before I can merge.

That being said there are gaps with the ability to test this lab effectively, as the main commands that would validate whether this works or not are being skipped. Given the timeline we're looking to get this in I'd be fine as long as we agree this can be addressed as a follow-up.

**Comment by niallthomson** [comment]
Thanks for addressing those changes @ybezsonov This will be merged on Monday as the scope for our release tomorrow is locked.

File: manifests/modules/automation/gitops/flux/.workshop/terraform/addon.tf
Comment (ybezsonov) [reply manifests/modules/automation/gitops/flux/.workshop/terraform/addon.tf]: I added changes from @csantanapr and random suffix for the ecr repository. I also enabled tests for all of the steps. It takes time to pass the tests and the last step need larger timeout, but we don't have skipped tests anymore.

**Pull request merged**
